VirtoCommerce.Platform.Data/Assets/AssetEntryService.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Caching.Memory;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.Platform.Core.Caching;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Data.Caching;
+using VirtoCommerce.Platform.Data.Repositories;
+
+namespace VirtoCommerce.Platform.Data.Assets
+{
+    public class AssetEntryService:  IAssetEntryService, IAssetEntrySearchService
+    {
+        private readonly Func<IPlatformRepository> _platformRepository;
+        private readonly IBlobUrlResolver _blobUrlResolver;
+        private readonly IMemoryCache _memoryCache;
+
+        public AssetEntryService(Func<IPlatformRepository> repositoryFactory, IBlobUrlResolver blobUrlResolver, IMemoryCache memoryCache)
+        {
+            _platformRepository = repositoryFactory;
+            _blobUrlResolver = blobUrlResolver;
+            _memoryCache = memoryCache;
+        }
+
+        public  GenericSearchResult<AssetEntry> SearchAssetEntries(AssetEntrySearchCriteria criteria)
+        {
+            var cacheKey = CacheKey.With(GetType(), "SearchAssetEntries", criteria.GetHashCode().ToString());
+
+            return _memoryCache.GetOrCreateExclusive(cacheKey, (cacheEntry) =>
```

Comment:
You should avoid usage caching there, it is not a critical method  for performance and to avoid problems with invalidation we can just avoid of caching code there